src/components/AddCardContainer.tsx

```diff
@@ -0,0 +1,287 @@
+import React, { ChangeEvent, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+import CardPreview from './CardPreview';
+import Input from './Input';
+import { useCardDispatch } from '../context/CardContext';
+import { isAlphabet, isNumber, regEx, validateMonth, validateYear } from '../utils/validateInput';
+import ErrorMessage from './ErrorMessage';
+import { getUniqueID } from '../utils/key';
+
+const AddCardContainer = () => {
+  const [cardNumbers, setCardNumbers] = useState<string[]>(['', '', '', '']);
+  const [cardExpirationDate, setCardExpirationDate] = useState<string[]>(['', '']);
+  const [cardOwner, setCardOwner] = useState<string[]>(['']);
+  const [cardCVC, setCardCVC] = useState<string[]>(['']);
+  const [cardPWD, setCardPWD] = useState<string[]>(['', '']);
+  const [expirationError, setExpirationError] = useState<boolean>(false);
+  const setCard = useCardDispatch();
+  const navigate = useNavigate();
+
+  const onChangeDateState = (e: ChangeEvent) => {
+    if (!(e.target instanceof HTMLInputElement)) return;
+
+    switch (e.target.dataset.id) {
+      case '0':
+        validateMonth(e.target);
+        break;
+      case '1':
+        validateYear(e.target);
+        break;
+      default:
+    }
+
+    onChangeState(cardExpirationDate, setCardExpirationDate, 'number')(e);
+  };
+
+  const onChangeState =
+    (state: string[], setState: React.Dispatch<React.SetStateAction<string[]>>, type: string) =>
+    (e: ChangeEvent) => {
+      if (!(e.target instanceof HTMLInputElement)) return;
```

Comment:
```suggestion
    (e: ChangeEvent<HTMLInputElement>) => {
```

런타임에 체크하는 건 저도 선호하는방식이긴 한데,
요 경우에는 굳이 필요없을거 같아요

참고로 이벤트 핸들러 함수는 `ChangeEventHandler<T> ` 로도 많이 선언됩니다~!

Reply:
확실히 input element에만 쓰이기 때문에 런타임에 체크할 필요가 없군요!
런타임 체크하는 로직 모두 변경하였습니다!